### PR TITLE
Trigon drive

### PR DESCRIPTION
**Pull request by Mboesn**

### CODE REVIEW

File: src/main/java/frc/robot/constants/RobotConstants.java
Comment (EliyaFishman) [inline src/main/java/frc/robot/constants/RobotConstants.java]: Remove all x lock occurrences. We use commands that require drivetrain instead of x locking

File: src/main/java/frc/robot/constants/robots/RobotA.java
Comment (EliyaFishman) [inline src/main/java/frc/robot/constants/robots/RobotA.java]: remove

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (EliyaFishman) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: remove

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (EliyaFishman) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: remove

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (EliyaFishman) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: remove

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (EliyaFishman) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: remove

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (EliyaFishman) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: change the name to `yInputCalculation`

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (EliyaFishman) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: move magic number to constant

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (EliyaFishman) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: for what this function exist?

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (EliyaFishman) [reply src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: for what this function exist?

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (EliyaFishman) [reply src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: rename to `xyInputCalculation`

**Review by EliyaFishman — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java:58]: }
 // remove this blank line
}

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: public void setThreshold(double threshold) {
and after it:
    public double getThreshold() {
and the same to the Sensetivity methods

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java:12]: // TODO: add documentation
remove

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java:13]: // remove blank line
    private double sensitivity;

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java:21]: double threshold) {
move up,  and yes I know that it is long

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: robotConstants.trigonDriveConstents.SENSITIVITY, robotConstants.trigonDriveConstents.THRESHOLD);
up

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java:27]: setSensitivity
would we need that?

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: Math.sqrt(yInput * yInput + xInput * xInput) < threshold || Math.abs(yInput) < Math.abs(xInput));
one tab less

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: xyInputCalculation
to 
xInputCalculation
AND 
uncomment

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: 0.25 to class final Y_LINEAR_THRESHOLD
2 to class final y_linear_coefficient (all caps)

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: // public double yInputCalculation(double value) {
uncomment

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: 0.5 to X_LINEAR_THRESHOLD
0.5 to x_linear_coefficient (all caps)

**Review by HilelAmiel — CHANGES_REQUESTED** [review]
more cr

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java:20]: too many blank lines

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: private static final double

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: add: double x = xInputCalculation(xInput);
double y = yInputCalculation(yInput);
and use them

**Review by HilelAmiel — CHANGES_REQUESTED** [review]
so close...

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (EliyaFishman) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java:20]: What happened to the second constructor?

**Review by EliyaFishman — CHANGES_REQUESTED** [review]
It's pretty good. But we still need to wait for drivetrain merge before this can be merged

**Review by EliyaFishman — APPROVED** [review]

**Review by EliyaFishman — COMMENTED** [review]
Good I will review drivetrain next

**Comment by EliyaFishman** [comment]
@HilelAmiel want to give it a quick look?

File: src/main/java/frc/robot/subsystems/drivetrain/Drivetrain.java
Comment (EliyaFishman) [inline src/main/java/frc/robot/subsystems/drivetrain/Drivetrain.java]: driv

File: src/main/java/frc/robot/subsystems/drivetrain/Drivetrain.java
Comment (EliyaFishman) [inline src/main/java/frc/robot/subsystems/drivetrain/Drivetrain.java]: driv

**Review by EliyaFishman — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/constants/RobotConstants.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/constants/RobotConstants.java:15]: public TrigonDriveConstents trigonDriveConstents = new TrigonDriveConstents();
move this after drivetrainConstants

File: src/main/java/frc/robot/constants/RobotConstants.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/constants/RobotConstants.java]: move this after DrivetrainConstants

File: src/main/java/frc/robot/constants/robots/RobotA.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/constants/robots/RobotA.java]: move after driveConstents
and set:
SENSITIVITY = 1
THRESHOLD = 0.5

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (HilelAmiel) [reply src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java:20]: we don't need it because we have default SENSITIVITY and if we want to change then use setSensitivity

File: src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java
Comment (HilelAmiel) [inline src/main/java/frc/robot/subsystems/drivetrain/TrigonDrive.java]: curvatureDrive(sensitivity * x, sensitivity * y,
change x and y positions

**Review by HilelAmiel — CHANGES_REQUESTED** [review]
one thing

**Review by HilelAmiel — APPROVED** [review]
finnaly

**Pull request merged**
